Assets/RGScripts/RGActionMap.cs

```diff
@@ -0,0 +1,21 @@
+/*
+* This file has been automatically generated. Do not modify.
+*/
+
+using System;
+
+namespace RegressionGames
+{
+    using UnityEngine;
+
+    public class RGActionMap : MonoBehaviour
+    {
+        private void Awake()
+        {
+            if (this.TryGetComponent<Player>(out var _))

```

Comment:
I'm still not sold on the ActionMap concept.

When you have a simple game with only 1 or 2 player classes, I can see where it gives you a single spot to configure all the actions without having to attach things in the editor.

That said, it also gives me no visibility in the editor as to what actions are attached to my object.


I'm wondering if instead of having just 1 RGActionMap, we could instead have multiple unique named actionLists.

Thus I could then associate `PlayerActionList` to my Player prefabs, or `ImpActionList` to my imps, etc.  These don't really need to be editable in the editor, but being able to see the list of actions for the associated type in the editor would be very helpful.

Taking BossRoom as an example though; this pattern probably still doesn't go far enough, as the player object is always the same, but the abilities are different/unique based on the player's assigned class (mage,rogue,archer,tank).

Reply:
Updated in docs 👍